### PR TITLE
Only left mouse button should be able to start or end drags

### PR DESCRIPTION
**Pull request by rickwebiii**

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by longlho** [comment]
hmm this is primarily for touch, not mouse.

**Comment by rickwebiii** [comment]
Touch backend optionally allows mouseevents if you pass mouseEvents:true when you create the backend. This change addresses mouse events when you do this.

**Comment by JoaoMosmann** [comment]
For me it's ok. But it should be configurable, by default, don't allow left click to drag. But if you wan't, should it have a way.

**Comment by rickwebiii** [comment]
The current behavior is as you describe; the backend only responds to mouseevents if you pass enableMouseEvents: true. Otherwise, you can only use touch to drag and drop with this backend. This change doesn't change that. See https://github.com/yahoo/react-dnd-touch-backend#mouse-events-support.

This change fixes a number of strange issues if you are using mouse events. For example, today you can currently left mouse down and center mousedown and up. This will end the drag, despite the left mouse button still being held. React-dnd exposes no way to know which mouse button you used to drag items and doesn't track independent drag states for multiple buttons. As such, there's no real reason to allow anything other than left button when you have mouse events enabled, as doing so just creates a class of bugs you can't fix.

File: src/Touch.js [hunk 1]
```diff
@@ -282,6 +296,11 @@ export class TouchBackend {
     }
 
     handleTopMoveStartCapture (e) {
+        // If the event is a MouseEvent, only the left mouse button should be able to start drags.
+        if (e.button === undefined && e.button !== MouseButton.Left) {
```
Comment (flakeparadigm) [inline src/Touch.js]: I think this logic might have a typo: if e.button is undefined AND e.button equals left mouse button.

Comment (JoaoMosmann) [inline src/Touch.js]: Well spotted!

Comment (rickwebiii) [reply src/Touch.js]: Yeah, I screwed that up. Moving the logic to functions for reuse.

**Comment by rickwebiii** [comment]
Sorry for the slow turnaround, but I got sidetracked for a while. Ping.

**Comment by flakeparadigm** [comment]
Seems good to me, though I ultimately don't have any power here. haha

**Comment by longlho** [comment]
published as 0.3.18

**Pull request merged**